tests/numerics/ad/test_operators.py

```diff
@@ -651,6 +629,170 @@ def _compare_ad_objects(a, b):
     assert np.allclose(true_state[ind1], v1_prev.value(eq_system, true_iterate))
 
 
+def test_ad_variable_prev_iter():
```

Comment:
Testing of arbitrary prev iter depth.
@keileg  Note that I changed the Variable.id usage: Prev iter and time have for now the same ID since the framework does not fully support changes in dofs.
This requires more thinking and work on the internal data structures.

Reply:
Noted.